src/commands/lambda/flare.ts

```diff
@@ -349,9 +349,16 @@ export class LambdaFlareCommand extends Command {
       }
 
       // Write insights file
-      const insightsFilePath = path.join(rootFolderPath, INSIGHTS_FILE_NAME)
-      generateInsightsFile(insightsFilePath, this.isDryRun, config)
-      this.context.stdout.write(`• Saved insights file to ./${INSIGHTS_FILE_NAME}\n`)
+      try {
+        const insightsFilePath = path.join(rootFolderPath, INSIGHTS_FILE_NAME)
+        generateInsightsFile(insightsFilePath, this.isDryRun, config)
+        this.context.stdout.write(`• Saved insights file to ./${INSIGHTS_FILE_NAME}\n`)
+      } catch (err) {
+        const errorDetails = err instanceof Error ? err.message : ''
+        this.context.stdout.write(
+          helpersRenderer.renderSoftWarning(`Unable to create INSIGHTS.md file. ${errorDetails}`)
+        )
+      }
```

Comment:
Adding an inner try/catch because we don't want to exit the program just because there was an error in the INSIGHTS.md generation.
Insights is a nice to have, but shouldn't block the flare program from running